Content.Server/GameObjects/EntitySystems/Weapon/Melee/FlashSystem.cs

```diff
@@ -0,0 +1,132 @@
+using Content.Shared.GameObjects.EntitySystems;
+using Content.Shared.Interfaces;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.GameObjects;
+using Robust.Shared.IoC;
+using Robust.Shared.Localization;
+using Robust.Shared.Player;
+
+namespace Content.Server.GameObjects.EntitySystems.Weapon.Melee
+{
+    public class FlashSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<FlashComponent, ClickAttackEvent>(OnClickAttack);
+            SubscribeLocalEvent<FlashComponent, UseInHandEvent>(OnUseInHand);
+
+            SubscribeLocalEvent<FlashComponent, ExaminedEvent>(OnExamined);
+        }
+
+        public void OnClickAttack(EntityUid uid, FlashComponent comp, ClickAttackEvent args)
+        {
+            if (!UseFlash(comp, args.User))
+            {
+                return;
+            }
+
+            if (args.TargetEntity != null)
+                FlashEntity(args.TargetEntity, args.User, comp.FlashDuration, comp.SlowTo);
+        }
+
+        public void OnUseInHand(EntityUid uid, FlashComponent comp, UseInHandEvent args)
+        {
+            if (!UseFlash(comp, args.User))
+            {
+                return;
+            }
+
+            foreach (var entity in IoCManager.Resolve<IEntityLookup>().GetEntitiesInRange(comp.Owner.Transform.Coordinates, comp.Range))
+            {
+                FlashEntity(entity, args.User, comp.AoeFlashDuration, comp.SlowTo);
+            }
+        }
+
+        private bool UseFlash(FlashComponent comp, IEntity user)
+        {
+            if (comp.HasUses)
+            {
+                // TODO flash visualizer
+                var sprite = comp.Owner.GetComponent<SpriteComponent>();
+                if (--comp.Uses == 0)
+                {
+                    sprite.LayerSetState(0, "burnt");
+                    comp.Owner.PopupMessage(user, Loc.GetString("flash-component-becomes-empty"));
+                }
+                else if (!comp.Flashing)
+                {
+                    int animLayer = sprite.AddLayerWithState("flashing");
+                    comp.Flashing = true;
+
+                    comp.Owner.SpawnTimer(400, () =>
+                    {
+                        sprite.RemoveLayer(animLayer);
+                        comp.Flashing = false;
+                    });
+                }
+
+                SoundSystem.Play(Filter.Pvs(comp.Owner), "/Audio/Weapons/flash.ogg", comp.Owner.Transform.Coordinates,
+                    AudioParams.Default);
+
+                return true;
+            }
+
+            return false;
+        }
+
+        // TODO: Check if target can be flashed (e.g. things like sunglasses would block a flash)
+        // TODO: Merge with the code in FlashableComponent--raise an event on the target, that FlashableComponent or
```

Comment:
not sure if I want to do this in this PR or in another PR

Reply:
You can do it in another PR, don't worry